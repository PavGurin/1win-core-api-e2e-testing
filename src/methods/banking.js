import {randomStr} from "../randomizer";

export const banking = {

    async transfet_create() {
        return await socket.send('BANKING:transfer-create', {
            targetEmail: randomStr(5) + '_transfet@test.xyz',
            amount: 100,
            currency: 'RUB'
        });
    },

    async withdrawal_create() {
        const result = await socket.send('BANKING:withdrawal-create', {
            amount: '100',
            wallet: '0000111122223333',
            payment_system: 'card_rub',
            currency: 'RUB'
        });
        //console.log(JSON.stringify(result, null, 2));
    },

}