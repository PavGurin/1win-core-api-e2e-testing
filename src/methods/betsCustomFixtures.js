/* eslint no-param-reassign: off */

import { mysqlConnection } from './mysqlConnection';
import { randomNum } from '../randomizer';
import { formatDateYyyyMmDdHhIiSs, rndNumInRange } from './utils';

async function getLastTranslationId() {
  const [res] = await mysqlConnection.executeQuery('select translate_id from 1win_test.custom_translations order by translate_id desc limit 1;', 'custom_bets');
  // console.log(res);
  return res.translate_id;
}

export const betsCustomFixtures = {
  async deafaultLangs() {
    return mysqlConnection.executeQuery('insert into 1win_test.custom_langs (id, value, title_ru, title_world, is_disabled) values(\'1\', \'en\', \'Английский\', \'English\', 0), (\'2\', \'ru\', \'Русский\', \'Russian\', 0);',
      'custom_bets');
  },

  async addLang(value, titleRu, titleEn, isDisabled) {
    return mysqlConnection.executeQuery(`insert into 1win_test.custom_langs(value, title_ru, title_world, is_disabled) 
     values (${value}, ${titleRu}, ${titleEn}, ${isDisabled})`, 'custom_bets');
  },

  async addGroup(titleRu, titleEn, createdAt = formatDateYyyyMmDdHhIiSs(new Date()),
    updatedAt = formatDateYyyyMmDdHhIiSs(new Date()), isDisabled = 0) {
    const rnd = randomNum(10);
    if (!titleRu) { titleRu = `название группы ${rnd}`; }
    if (!titleEn) { titleEn = `group title ${rnd}`; }

    const tid = await getLastTranslationId();

    await mysqlConnection.executeQuery(`insert into 1win_test.custom_translations (translate_id, value, lang_id, is_default) 
          values ('${tid + 1}', '${titleRu}', 2, 1), ('${tid + 1}', '${titleEn}', 1, 0);`, 'custom_bets');

    await mysqlConnection.executeQuery(`insert into 1win_test.custom_groups (title_id, created_at, updated_at, is_disabled, user_author_id) 
     values ('${tid + 1}','${createdAt}', '${updatedAt}', '${isDisabled}', '1001')`, 'custom_bets');

    const [groupId] = await mysqlConnection.executeQuery(`select id from 1win_test.custom_groups where created_at = '${createdAt}' and title_id = '${tid + 1}' ;`, 'custom_bets');
    return {
      titleRu, titleEn, createdAt, updatedAt, id: groupId.id,
    };
  },

  async addEvent(groupId, titleRu, titleEn, descriptionRu, descriptionEn,
    createdAt = formatDateYyyyMmDdHhIiSs(new Date()),
    updatedAt = formatDateYyyyMmDdHhIiSs(new Date()),
    startAt = formatDateYyyyMmDdHhIiSs(new Date()), stopAt, isDisabled = 0) {
    const rnd = randomNum(10);
    if (!titleRu) { titleRu = `название события ${rnd}`; }
    if (!titleEn) { titleEn = `event title ${rnd}`; }
    if (!descriptionRu) { descriptionRu = `описание события ${rnd}`; }
    if (!descriptionEn) { descriptionEn = `event description ${rnd}`; }
    if (!stopAt) {
      stopAt = new Date();
      stopAt.setDate(stopAt.getDate() + 10);
      stopAt = formatDateYyyyMmDdHhIiSs(stopAt);
    }

    const tid = await getLastTranslationId();

    await mysqlConnection.executeQuery(`insert into 1win_test.custom_translations (translate_id, value, lang_id, is_default)
           values ('${tid + 1}', '${titleRu}', 2, 1), ('${tid + 1}', '${titleEn}', 1, 0),
                  ('${tid + 2}', '${descriptionRu}', 2, 1), ('${tid + 2}', '${descriptionEn}', 1, 0)`, 'custom_bets');

    await mysqlConnection.executeQuery(`insert into 1win_test.custom_events (group_id, title_id, created_at, updated_at, 
                                     user_author_id, description_id, start_at, stop_at, is_calculated, is_disabled)
        values ('${groupId}', '${tid + 1}', '${createdAt}', '${updatedAt}', 
                '1001', '${tid + 2}', '${startAt}', '${stopAt}', '0', '${isDisabled}')`, 'custom_bets');

    const [eventId] = await mysqlConnection.executeQuery(`select id from 1win_test.custom_events where created_at = '${createdAt}' and title_id = '${tid + 1}' ;`, 'custom_bets');
    return { // eslint-disable-next-line max-len
      groupId, titleRu, titleEn, descriptionRu, descriptionEn, createdAt, updatedAt, startAt, stopAt, id: eventId.id,
    };
  },

  async addResult(eventId, titleRu, titleEn, descriptionRu, descriptionEn,
    factor = rndNumInRange(1, 10).toFixed(2), isDisabled = 0, outcome = 0,
    createdAt = formatDateYyyyMmDdHhIiSs(new Date()),
    updatedAt = formatDateYyyyMmDdHhIiSs(new Date())) {
    const rnd = randomNum(10);
    if (!titleRu) { titleRu = `название исхода ${rnd}`; }
    if (!titleEn) { titleEn = `result title ${rnd}`; }
    if (!descriptionRu) { descriptionRu = `описание исхода ${rnd}`; }
    if (!descriptionEn) { descriptionEn = `result description ${rnd}`; }

    const tid = await getLastTranslationId();

    await mysqlConnection.executeQuery(`insert into 1win_test.custom_translations (translate_id, value, lang_id, is_default)
           values ('${tid + 1}', '${titleRu}', 2, 1), ('${tid + 1}', '${titleEn}', 1, 0),
                  ('${tid + 2}', '${descriptionRu}', 2, 1), ('${tid + 2}', '${descriptionEn}', 1, 0)`, 'custom_bets');

    await mysqlConnection.executeQuery(`insert into 1win_test.custom_results (event_id, title_id, user_author_id,
                                      factor, is_disabled, outcome, created_at, updated_at, description_id)
        values ('${eventId}', '${tid + 1}', '1001', '${factor}', '${isDisabled}', '${outcome}', '${createdAt}', 
                '${updatedAt}', ${tid + 2})`, 'custom_bets');

    const [resultId] = await mysqlConnection.executeQuery(`select id from 1win_test.custom_results where created_at = '${createdAt}' and factor = '${factor}' ;`, 'custom_bets');
    return {
      eventId,
      titleRu,
      titleEn,
      descriptionRu,
      descriptionEn,
      factor,
      outcome,
      createdAt,
      updatedAt,
      id: resultId.id,
    };
  },

  async createGroupWithEventWithResults(resultsNumber) {
    const group = await this.addGroup();
    const event = await this.addEvent(group.id);
    const num = [...Array(resultsNumber).keys()];
    const results = [];
    for await (const i of num) { // eslint-disable-line no-restricted-syntax
      results.push(await this.addResult(event.id)); // eslint-disable-line no-await-in-loop
    }

    return { group, event, results };
  },

  async setGroupIsDisabled(groupId, isDisabled) {
    return mysqlConnection.executeQuery(`update 1win_test.custom_groups set is_disabled = '${isDisabled}' where id = '${groupId}';`, 'custom_bets');
  },

  async setEventIsDisabled(eventId, isDisabled) {
    return mysqlConnection.executeQuery(`update 1win_test.custom_events set is_disabled = '${isDisabled}' where id = '${eventId}';`, 'custom_bets');
  },

  async setResultIsDisabled(resultId, isDisabled) {
    return mysqlConnection.executeQuery(`update 1win_test.custom_results set is_disabled = '${isDisabled}' where id = '${resultId}';`, 'custom_bets');
  },

  async setEventCalculated(eventId, isCalculated) {
    return mysqlConnection.executeQuery(`update 1win_test.custom_events set is_calculated = '${isCalculated}' where id = '${eventId}';`, 'custom_bets');
  },

  async setEventStopTime(eventId, stopTime) {
    return mysqlConnection.executeQuery(`update 1win_test.custom_events set stop_at = '${stopTime}' where id = '${eventId}';`, 'custom_bets');
  },
};
