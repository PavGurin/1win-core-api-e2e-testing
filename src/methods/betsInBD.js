import { mysqlConnection } from './mysqlConnection';
import { sleep } from './utils';

// статусы ставки
// 0 = ожидание
// 1 = проиграл
// 2 = выиграл
// 3 = возврат
// 4 = продана

// type, timeOpen, timeClose, device - можно не указывать
export async function addBetToBD(userId, currency, amount, coeff, status,
  type = 'ordinary', timeOpen = new Date() / 1, timeClose = new Date() / 1, device = 'app-android') {
  const profit = status === 2 ? (amount * coeff).toFixed(2) : 0;
  await mysqlConnection.executeQuery(`INSERT INTO 1win.ma_bets(id_user, time_open, time_close, 
                         status, amount, total_coefficient, currency, sys, flags, profit, promo_amount, 
                         betType, device) 
                         VALUES(${userId}, '${timeOpen}', ${timeClose}, ${status}, ${amount}, ${coeff}, 
                                '${currency}', 0, 0, ${profit}, 0, '${type}', '${device}')`);
}

export async function addSoldBetToBD(userId, currency, amount, profit, coeff, type = 'ordinary', timeOpen = new Date() / 1, timeClose = new Date() / 1, device = 'app-android') {
  await mysqlConnection.executeQuery(`INSERT INTO 1win.ma_bets(id_user, time_open, time_close, 
                         status, amount, total_coefficient, currency, sys, flags, profit, promo_amount, 
                         betType, device) 
                         VALUES('${userId}', '${timeOpen}', '${timeClose}', '4', '${amount}', '${coeff}', 
                                '${currency}', '0', '0', '${profit}', '0', '${type}', '${device}')`);
}


export async function getUsersLastBetId(userId) {
  const result = await mysqlConnection.executeQuery(`SELECT id FROM 1win.ma_bets WHERE id_user = ${userId} ORDER BY id DESC LIMIT 1`);
  return result[0].id;
}

export async function getRandomSelection() {
  const selection = await mysqlConnection.executeQuery('SELECT * FROM 1win.ma_selections ORDER BY RAND() LIMIT 1');
  return selection[0];
}

export async function addSelectionToBD(betId, market, matchId, awayTeamId, homeTeamId,
  oddsId, typeId, subType, banker, ways, type,
  odd, line, freeText, matchName, status, voidFactor, description, mtsTicketId, betWinProportion) {
  await mysqlConnection.executeQuery(`INSERT INTO 1win.ma_selections(bet_id, market, match_id, awayteam_id, 
                               hometeam_id, odds_id, type_id, sub_type, banker, ways, type, odd, line, 
                               freetext, match_name, status, void_factor, description, mts_ticket_id, bet_win_proportion) 
                               VALUES ('${betId}', '${market}', '${matchId}', '${awayTeamId}', '${homeTeamId}', '${oddsId}', '${typeId}',
                                       '${subType}', '${banker}', '${ways}', '${type}', '${odd}', '${line}', '${freeText}', 
                                       '${matchName}', '${status}', '${voidFactor}', '${description}', '${mtsTicketId}', 
                                       '${betWinProportion}');`);
}

export async function getSelectionsByBetId(betId) {
  return mysqlConnection.executeQuery(`SELECT id from 1win.ma_selections WHERE bet_id = ${betId};`);
}

export async function setSelectionStatus(selectionId, newStatus) {
  await mysqlConnection.executeQuery(`UPDATE 1win.ma_selections SET status = '${newStatus}' WHERE id = ${selectionId}`);
}

export async function getBet(betId) {
  const result = await mysqlConnection.executeQuery(`SELECT * FROM 1win.ma_bets WHERE id = ${betId};`);
  return result[0];
}

export async function makeSuccessfulOrdinaryBet(user, currency, amount, coeff) {
  // добавляем ставку для юзера в ma_bets
  await addBetToBD(user.id, currency, amount, coeff, 0, 'ordinary');
  const betId = await getUsersLastBetId(user.id);
  // console.log(betId);

  // копируем рандомный селекшн в ma_selections и привязываем его к ставке
  const selection = await getRandomSelection();
  // selection.description.replace(/.*("coefficient":.*..*)}}.*/, `"coefficient":${coeff}`);
  // console.log(selection);
  await addSelectionToBD(betId, selection.market, selection.match_id, selection.awayteam_id,
    selection.hometeam_id, selection.odds_id, selection.type_id, selection.sub_type,
    selection.banker, selection.ways, selection.type,
    coeff, selection.line, selection.freetext, selection.match_name, 0, selection.void_factor,
    selection.description, selection.mts_ticket_id, selection.bet_win_proportion);
  const selectionId = await getSelectionsByBetId(betId);
  // console.log(selectionId);

  // выставляем селекшену статус = выигрыш
  await setSelectionStatus(selectionId[0].id, 2);

  // ждем, пока отработает скрипт, проставляющий ставке статус такой, как у ее селекшена
  await sleep(60000);

  // достаем из базы ставку
  const bet = await getBet(betId);
  return bet;
}

export async function makeSuccessfulExpressBet(user, currency, amount, coeff) {
  // добавляем ставку для юзера в ma_bets
  await addBetToBD(user.id, currency, amount, coeff, 0, 'express');
  const betId = await getUsersLastBetId(user.id);
  // console.log(betId);

  const coeffs = [1.1, 2, Math.round(coeff / 1.1 / 2 * 100) / 100];

  for (let i = 0; i < 3; i++) {
    /* eslint no-await-in-loop:off */
    // копируем рандомный селекшн в ma_selections и привязываем его к ставке
    const selection = await getRandomSelection();
    // selection.description.replace(/.*("coefficient":.*..*)}}.*/, `"coefficient":${coeff}`);
    // console.log(selection);
    await addSelectionToBD(betId, selection.market, selection.match_id, selection.awayteam_id,
      selection.hometeam_id, selection.odds_id, selection.type_id, selection.sub_type,
      selection.banker, selection.ways, selection.type,
      coeffs[i], selection.line, selection.freetext, selection.match_name, 0, selection.void_factor,
      selection.description, selection.mts_ticket_id, selection.bet_win_proportion);
  }

  const selectionsIds = await getSelectionsByBetId(betId);
  // console.log(selectionsIds);

  // выставляем селекшенам статус = выигрыш
  selectionsIds.forEach(async (selection) => {
    await setSelectionStatus(selection.id, 2);
  });

  // ждем, пока отработает скрипт, проставляющий ставке статус такой, как у ее селекшена
  await sleep(60000);

  // достаем из базы ставку
  const bet = await getBet(betId);
  return bet;
}
