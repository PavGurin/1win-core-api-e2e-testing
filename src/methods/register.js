import {randomNum, randomStr} from '../randomizer';

const partner_key = 'test001';

export const register = {

    async one_click_reg() {
        return await socket.send('USER:auth-register',
            {
                isShort: true,
                country: default_country,
                timezone: 23,
                partner_key,
            });
    },

    async usual_reg(usualRegistration) {
        return await socket.send('USER:auth-register',
            {
                isShort: false,
                name: randomStr(),
                email: `${randomStr(5)}_test@new.xyz`,
                phone: randomNum().toString(),
                password: default_password,
                repeat_password: default_password,
                country: default_country,
                timezone: 23,
                birthday: 946587600000,
                partner_key,
                ...usualRegistration,
            });
    },
};
