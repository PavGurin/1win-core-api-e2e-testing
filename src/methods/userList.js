/* eslint camelcase: 'off' */
import { randomStr } from '../randomizer';

export const userList = {

  async loginWithParams(login, password, tg_hash) {
    return socket.send('USER:auth-login', {
      login,
      password,
      tg_hash,
    });
  },

  // Prodlike

  async loginWithRub() {
    return socket.send('USER:auth-login', {
      login: 'test_withdrawal@mailinator.com',
      password: '123123',
      tg_hash: randomStr(5),
    });
  },

  async loginWithRubUsd() {
    // Должен быть баланс в валюте отличной от рублей
    return socket.send('USER:auth-login', {
      login: 'test_withdrawal2@mailinator.com',
      password: '123123',
      tg_hash: randomStr(5),
    });
  },

  // С реальными деньгами! Использовать аккуратно)
  async loginWithRealMoney() {
    return socket.send('USER:auth-login', {
      login: 'nogm75@1win.xyz',
      password: '123456',
    });
  },

  async loginTestStatus() {
    // У пользователя статит статус тестового пользователя
    return socket.send('USER:auth-login', {
      login: 'tester_status@mailinator.com',
      password: '123123',
      tg_hash: randomStr(5),
    });
  },

  async loginFullBlock() {
    // У пользователя стоит полный блок
    return socket.send('USER:auth-login', {
      login: 'full_block_user@mailinator.com',
      password: '123123',
      tg_hash: randomStr(5),
    });
  },

  async loginPartialBlock() {
    // У пользователя стоит частичный блок
    return socket.send('USER:auth-login', {
      login: 'partial_block_user@mailinator.com',
      password: '123123',
      tg_hash: randomStr(5),
    });
  },

  async loginManualControl() {
    // У пользователя стоит частичный блок
    return socket.send('USER:auth-login', {
      login: 'partial_block_user@mailinator.com',
      password: '123123',
      tg_hash: randomStr(5),
    });
  },
};
