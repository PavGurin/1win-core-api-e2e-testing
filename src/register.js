const partner_key = 'test001';
const birthday = 946587600002;
const default_password = '123456';

export const register = {

    async one_click_reg() {
        return await socket.send('USER:auth-register',
            {
                isShort: true,
                country: 'ru',
                timezone: 23,
                partner_key: partner_key
            });
    }
};

