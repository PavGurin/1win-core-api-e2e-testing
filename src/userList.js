import {randomStr} from './randomizer';

export const userList = {

    //Prodlike
    async login_without_money() {
        return await socket.send('USER:auth-login', {
            login: '123123@mailinator.com',
            password: '123123'
        });
    },

    async login_without_money_tghash() {
        return await socket.send('USER:auth-login', {
            login: '123123@mailinator.com',
            password: '123123',
            tg_hash: randomStr(5)
        });
    },

    async login_by_phone() {
        return await socket.send('USER:auth-login', {
            login: '+79995654567',
            password: '123123'
        });
    },

    async login_with_RUB() {
        return await socket.send('USER:auth-login', {
            login: 'test_withdrawal@mailinator.com',
            password: '123123',
            tg_hash: randomStr(5)
        });
    },

    async login_with_RUB_USD() {
        //Должен быть баланс в валюте отличной от рублей
        await socket.send('USER:auth-login', {
            login: 'test_withdrawal2@mailinator.com',
            password: '123123',
            tg_hash: randomStr(5)
        });
    },

    async login_test_status() {
        //У пользователя статит статус тестового пользователя
        await socket.send('USER:auth-login', {
            login: 'tester_status@mailinator.com',
            password: '123123',
            tg_hash: randomStr(5)
        });
    },

    async login_full_block() {
        //У пользователя стоит полный блок
        await socket.send('USER:auth-login', {
            login: 'full_block_user@mailinator.com',
            password: '123123',
            tg_hash: randomStr(5)
        });
    },

    async login_partial_block() {
        //У пользователя стоит частичный блок
        await socket.send('USER:auth-login', {
            login: 'partial_block_user@mailinator.com',
            password: '123123',
            tg_hash: randomStr(5)
        });
    }
};

