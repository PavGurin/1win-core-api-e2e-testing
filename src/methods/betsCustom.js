/**
 *  ДОКА
 *  https://fbet-gitlab.ex2b.co/backend/proposals/issues/2
 */


/* eslint object-shorthand: off */
import { randomNum, randomStr } from '../randomizer';
import { formatDateYyyyMmDdHhIiSs, sleep } from './utils';
import { betsCustomFixtures } from './betsCustomFixtures';

export const betsCustom = {
  /**
   * клиентские запросы
   */
  async langs() {
    return socket.send('MATCH-STORAGE-CUSTOM:match-storage-custom-langs', {});
  },

  async groups() {
    return socket.send('MATCH-STORAGE-CUSTOM:match-storage-custom-groups', {});
  },

  async groupEvents(groupId, withResults = 0) {
    return socket.send('MATCH-STORAGE-CUSTOM:match-storage-custom-events', {
      groupId: groupId,
      withResults: withResults,
    });
  },

  async event(eventId) {
    return socket.send('MATCH-STORAGE-CUSTOM:match-storage-custom-event', {
      id: eventId,
    });
  },

  async makeBetShort(id, amount) {
    return socket.send('BETS:bets-make', {
      betsMap: {
        someString: {
          amount,
          couponList: [
            {
              id,
              providerId: 3,
            },
          ],
        },
      },
    });
  },

  async makeOrdinaryBet(resultId, amount, coeff) {
    return socket.send('BETS:bets-make', {
      betsMap: {
        someString: {
          amount,
          couponList: [
            {
              id: resultId,
              coefficient: coeff,
              providerId: 3,
            },
          ],
        },
      },
    });
  },

  async makeExpressBet(resultIdCoeffArray, amount) {
    const couponList = [];
    resultIdCoeffArray.forEach((item) => {
      couponList.push({
        id: item.id,
        coefficient: item.coeff,
        providerId: 3,
      });
    });
    return socket.send('BETS:bets-make', {
      betsMap: {
        someString: {
          amount,
          couponList: couponList,
        },
      },
    });
  },

  async successfulOrdinaryBet(eventId, results, amount) {
    let stop = new Date();
    stop.setSeconds(stop.getSeconds() + 10);
    stop = formatDateYyyyMmDdHhIiSs(stop, true);

    await betsCustomFixtures.setEventStopTime(eventId, stop);

    const { data } = await this.makeOrdinaryBet(results[0].id, amount, results[0].factor);
    // console.log(data);

    await sleep(10000);
    await betsCustomFixtures.setResultOutcome(results[0].id, 2);
    await betsCustomFixtures.setResultOutcome(results[1].id, 1);
    await sleep(45000);
  },

  async maxBetAmount(resultId, coeff) {
    return socket.send('BETS:bets-maxBetAmount', {
      couponList: [
        {
          id: resultId,
          providerId: 3,
          coefficient: coeff,
          match: {
            categoryId: 0,
            matchId: 0,
            sportId: 0,
            tournamentId: 0,
          },
          odd: {
            coefficient: '',
            outCome: '',
            service: '',
            specialValue: '',
            subTypeId: 0,
            typeId: 0,
          },
        },
      ],
    });
  },

  /**
   * запросы для админки
   * ! не работают,  если на стейдже не отключена проверка на транспорт, ошибка
   * { status: 403, message: 'Access restricted for this transport' }
   */
  async langsAdmin() {
    return socket.send('MATCH-STORAGE-CUSTOM:match-storage-custom-admin-langs', {});
  },

  async groupsAdmin() {
    return socket.send('MATCH-STORAGE-CUSTOM:match-storage-custom-admin-groups', {});
  },

  async groupEventsAdmin(groupId) {
    return socket.send('MATCH-STORAGE-CUSTOM:match-storage-custom-admin-events', {
      groupId: groupId,
    });
  },

  async eventAdmin(eventId) {
    return socket.send('MATCH-STORAGE-CUSTOM:match-storage-custom-admin-event', {
      id: eventId,
    });
  },

  async addGroup(groupNameLangArray) {
    return socket.send('MATCH-STORAGE-CUSTOM:match-storage-custom-admin-groups-add', {
      title: groupNameLangArray,
    });
  },

  async getGroupByTitle(groupTitle) {
    const { data: groups } = await this.groupsAdmin();
    // console.log(groups);
    return groups.find(receivedGroup => receivedGroup.titles
      .some(title => title.value === groupTitle));
  },

  async getGroupById(groupId) {
    const { data: groups } = await this.groupsAdmin();
    return groups.find(receivedGroup => receivedGroup.id === groupId);
  },

  async createGroup(group) {
    if (!group) {
      // eslint-disable-next-line no-param-reassign
      group = [{
        value: `auto group ${randomNum(10)}`,
        langId: 1,
        isDefault: true,
      }, {
        value: `авто группа ${randomNum(10)}`,
        langId: 2,
      },
      ];
    }
    await this.addGroup(group);
    return this.getGroupByTitle(group[0].value);
  },

  async changeGroup(groupId, groupNameLangArray, isDisabled) {
    return socket.send('MATCH-STORAGE-CUSTOM:match-storage-custom-admin-groups-change', {
      id: groupId,
      title: groupNameLangArray,
      isDisabled: isDisabled,
    });
  },

  async addEvent(groupId, titlesArray, descriptionsArray, start, stop, resultsArray) {
    return socket.send('MATCH-STORAGE-CUSTOM:match-storage-custom-admin-events-add', {
      groupId: groupId,
      title: titlesArray,
      description: descriptionsArray,
      start: start,
      stop: stop,
      results: resultsArray,
    });
  },

  async allEvents(groupId) {
    const { data } = await socket.send('MATCH-STORAGE-CUSTOM:match-storage-custom-admin-events', {
      groupId,
    });
    return data;
  },

  async addEventHardcode() {
    const data = await socket.send('MATCH-STORAGE-CUSTOM:match-storage-custom-admin-events-add', {
      groupId: 2, // группа исходов
      title: [
        {
          value: 'Успешный деплой',
          langId: 2,
          isDefault: 1, // Обязательный флаг у одного из переводов
        },
        {
          value: 'Successful deploy',
          langId: 1,
        },
      ],
      description: [
        {
          value: 'Пройдет ли деплой без багов',
          langId: 2,
          isDefault: 1, // Обязательный флаг у одного из переводов
        },
        {
          value: 'Will the deploy pass without bugs',
          langId: 1,
        },
      ],
      start: '2020-02-06 11:22:45', // дата старта события строкой такого формата
      stop: '2020-02-29 15:15:36', // дата завершения события строкой такого формата, можно не указывать сразу
      results: [
        {
          title: [
            {
              value: 'Да',
              langId: 2,
              isDefault: 1, // Обязательный флаг у одного из переводов
            },
            {
              value: 'Yes',
              langId: 1,
            },
          ],
          description: [
            {
              value: 'Багов нет',
              langId: 2,
              isDefault: 1, // Обязательный флаг у одного из переводов
            },
            {
              value: 'There are no bugs',
              langId: 1,
            },
          ],
          factor: '10',
        }],
    });
    // console.log(data);
    return data;
  },

  async getEventByTitle(eventGroupId, eventTitle) {
    const { data: events } = await this.groupEventsAdmin(eventGroupId);
    const result = events.find(event => event.titles.some(title => title.value === eventTitle));
    return this.eventAdmin(result.id);
  },

  async createEvent(groupId, eventTitles, eventDescriptions, start, stop, eventResults) {
    /* eslint no-param-reassign:off */
    if (!eventTitles) {
      eventTitles = [{
        value: `auto event ${randomNum(10)}`,
        langId: 1,
        isDefault: true,
      }, {
        value: `авто событие ${randomNum(10)}`,
        langId: 2,
      },
      ];
    }

    if (!eventDescriptions) {
      eventDescriptions = [{
        value: `auto event description ${randomNum(10)}`,
        langId: 1,
        isDefault: true,
      },
      {
        value: `авто описание события ${randomNum(10)}`,
        langId: 2,
      },
      ];
    }

    if (!start) {
      start = formatDateYyyyMmDdHhIiSs(new Date());
    }

    if (!stop) {
      stop = formatDateYyyyMmDdHhIiSs(new Date());
    }

    if (!eventResults) {
      eventResults = [{
        title: [{
          value: `auto result 1 ${randomNum(10)}`,
          langId: 1,
          isDefault: true,
        }, {
          value: `авто исход 1 ${randomNum(10)}`,
          langId: 2,
        }],
        description: [{
          value: `auto result description 1 ${randomNum(10)}`,
          langId: 1,
          isDefault: true,
        }, {
          value: `авто описание исхода 1 ${randomNum(10)}`,
          langId: 2,
        }],
        factor: (Math.random() * 10).toFixed(2),
      }, {
        title: [{
          value: `auto result 2 ${randomNum(10)}`,
          langId: 1,
          isDefault: true,
        }, {
          value: `авто исход 2 ${randomNum(10)}`,
          langId: 2,
        }],
        description: [{
          value: `auto result description 2 ${randomNum(10)}`,
          langId: 1,
          isDefault: true,
        }, {
          value: `авто описание исхода 2 ${randomNum(10)}`,
          langId: 2,
        }],
        factor: (Math.random() * 10).toFixed(2),
      }];
    }

    await this.addEvent(groupId, eventTitles, eventDescriptions, start, stop, eventResults);
    // await sleep(500);
    const { data: event } = await this.getEventByTitle(groupId, eventTitles[0].value);
    return event;
  },

  async changeEvent(eventId, title, description, start, stop, isDisabled) {
    return socket.send('MATCH-STORAGE-CUSTOM:match-storage-custom-admin-events-change', {
      id: eventId,
      title: title,
      description: description,
      start: start,
      stop: stop,
      isDisabled: isDisabled,
    });
  },

  async changeResult(resultId, titleArray, descriptionArray, factor, outcome,
    isDisabled) {
    return socket.send('MATCH-STORAGE-CUSTOM:match-storage-custom-admin-results-change', {
      id: resultId,
      title: titleArray,
      description: descriptionArray,
      factor: factor,
      outcome: outcome,
      isDisabled: isDisabled,
    });
  },

  async unfinishedEvents(count) {
    return socket.send('MATCH-STORAGE-CUSTOM:match-storage-custom-admin-events-unfinished', {
      count: count,
    });
  },

  async finishEvents(eventIdsArray) {
    return socket.send('MATCH-STORAGE-CUSTOM:match-storage-custom-admin-events-finish', {
      events: eventIdsArray,
    });
  },

  async eventsBulk(eventIdsArray) {
    return socket.send('MATCH-STORAGE-CUSTOM:match-storage-custom-admin-events-bulk', {
      ids: eventIdsArray,
    });
  },


  async resultsBulk(resultIdsArray) {
    return socket.send('MATCH-STORAGE-CUSTOM:match-storage-custom-admin-results-bulk', {
      ids: resultIdsArray,
    });
  },
};
