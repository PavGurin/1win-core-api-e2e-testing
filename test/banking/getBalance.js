import { userList } from '../../src/methods/userList';
import { register } from '../../src/methods/register';

describe('Balance get', () => {
  it('C19353 (+) Without money', async () => {
    await register.oneClickReg(socket);
    const { data } = await socket.send('BANKING:balance-get');
    // console.log(data);
    expect(data).toMatchSnapshot();
  });

  it('C19354 (+) With money rub ', async () => {
    await userList.loginWithRub(socket);
    const { data } = await socket.send('BANKING:balance-get');
    // console.log(data);
    expect(data['0'].balance).not.equal(0);
    expect(data['0'].currency).equal('RUB');
    expect(data['1'].balance).not.equal(0);
    expect(data['1'].currency).equal('USD');
    expect(data['2'].balance).equal(0);
    expect(data['2'].currency).equal('EUR');
  });

  it('C21429 (+) Balance USD, RUB, EUR > 0', async () => {
    await userList.loginWithRealMoney(socket);
    const { data } = await socket.send('BANKING:balance-get');
    // console.log(data);
    expect(data).toMatchSnapshot();
  });
});
