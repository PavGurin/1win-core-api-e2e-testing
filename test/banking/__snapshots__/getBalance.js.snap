// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Balance get C19353 (+) Without money 1`] = `
Array [
  Object {
    "balance": 0,
    "currency": "RUB",
  },
  Object {
    "balance": 0,
    "currency": "USD",
  },
  Object {
    "balance": 0,
    "currency": "EUR",
  },
]
`;
